trin-core/src/portalnet/protocol.rs

```diff
@@ -241,12 +232,12 @@ impl PortalnetProtocol {
             db,
         };
 
-        let rpc_handler = JsonRpcHandler {
-            discovery,
-            jsonrpc_rx,
+        let proto = Self {
+            discovery: discovery.clone(),
+            overlay: overlay.clone(),
```

Comment:
Was there anything important about the move of `let proto`? It's helpful to me as a reviewer when a PR avoids moving code if the move has no effect, to make the diff smaller and quicker to review.

Reply:
That's right, I will refrain from such moves in the future.